10/31 node bbe8888

Backtest result: 

Only BTCUSDT attains 55.8% winrate

ETH, DOGE performs only 50%.

Explore the reasons

- The winrate of BTCUSDT is significantly higher than that of other cryptocurrencies.

In [1]:
import os, sys
sys.path.append("/Users/user/Desktop/stock/CryptoPredictor")
import pipeline

In [2]:
DELAY = 1000
HOURS = 72
SPLIT_RATE = 0.8

targets = ["BTCUSDT", "ETHUSDT","BNBUSDT", "SOLUSDT", "XRPUSDT", "DOGEUSDT"]
for t in targets:
    pipeline.backtest(reset=True, target=t)
    


remove all results
backtest on BTCUSDT
selected factors:
- weilun01              with number of config: 4
- weilun02              with number of config: 12
- weilun03              with number of config: 9
- weilun04              with number of config: 10
- weilun05              with number of config: 10
- Cross                 with number of config: 6
- Slope                 with number of config: 8
- Skewness              with number of config: 7
- CrossRSI              with number of config: 6
total number of factors: 72


factors: 100%|██████████| 72/72 [03:30<00:00,  2.92s/it]


              train corr  test corr   weights
Slope_200      -0.107080   0.318304 -0.031663
Slope_100      -0.067873  -0.020983 -0.016630
Slope_300      -0.059413   0.342480 -0.027929
weilun04_100   -0.058629  -0.003614 -0.047224
Skewness_10    -0.046608   0.108589 -0.003842
...                  ...        ...       ...
Skewness_200    0.083407   0.163454  0.005011
Skewness_50     0.088796   0.001548  0.005456
Skewness_100    0.089733   0.020948  0.005188
weilun03_200    0.092705   0.040802  0.050919
weilun03_400    0.107799   0.125773  0.020340

[70 rows x 3 columns]

train result:
r2_score 0.10722955572766724
accuracy 0.6170157068062827
correlation 0.3299114050758067
confusion
[[1325  690]
 [ 773 1032]]

test result:
r2_score -0.43746893527063024
accuracy 0.5581151832460733
correlation -0.0421599148793677
confusion
[[229 171]
 [251 304]]
remove all results
backtest on ETHUSDT
selected factors:
- weilun01              with number of config: 4
- weilun02              with number of con

factors: 100%|██████████| 72/72 [04:25<00:00,  3.69s/it]


                   train corr  test corr   weights
weilun03_300        -0.096649  -0.166068 -0.076543
weilun03_400        -0.084763  -0.161290 -0.003808
weilun04_100        -0.041298   0.083003 -0.064990
Cross_3_8           -0.038760   0.162649 -0.013209
Cross_5_10          -0.038744   0.167089 -0.014958
...                       ...        ...       ...
CrossRSI 5 15 10     0.054282   0.081026  0.018786
weilun03_20          0.056138  -0.110276  0.009999
weilun03_100         0.057871  -0.166175  0.015238
Skewness_100         0.069734  -0.117642  0.009926
Skewness_50          0.085049  -0.056654  0.008107

[70 rows x 3 columns]

train result:
r2_score 0.11333420584312637
accuracy 0.6178010471204188
correlation 0.3397036528636724
confusion
[[1241  664]
 [ 796 1119]]

test result:
r2_score -0.8391419070508535
accuracy 0.49947643979057593
correlation -0.15082957679673636
confusion
[[149 309]
 [169 328]]
remove all results
backtest on BNBUSDT
selected factors:
- weilun01              with n

factors:  71%|███████   | 51/72 [03:02<00:17,  1.19it/s]/Users/user/Desktop/stock/CryptoPredictor/pipeline.py:116: RankWarning: Polyfit may be poorly conditioned
  signal_output = pd.Series([fctr.Gen(row) for row in X], index=index, name=str(fctr))
factors: 100%|██████████| 72/72 [04:11<00:00,  3.49s/it]


              train corr  test corr   weights
weilun03_400   -0.087097  -0.135193 -0.033676
weilun03_300   -0.076756  -0.187170  0.001515
weilun04_100   -0.062059   0.207043 -0.082573
weilun05_50    -0.054066  -0.020219 -0.003010
weilun04_50    -0.044012   0.142016 -0.036460
...                  ...        ...       ...
weilun03_20     0.069544  -0.186860  0.011870
Slope_100       0.085349   0.154051  0.002326
Skewness_20     0.086752   0.001323 -0.000391
Skewness_100    0.130826  -0.084856  0.008789
Skewness_50     0.155668   0.085052  0.006932

[70 rows x 3 columns]

train result:
r2_score 0.11950856302514623
accuracy 0.6104712041884817
correlation 0.3475165376860666
confusion
[[1324  666]
 [ 822 1008]]

test result:
r2_score -0.47457457572667106
accuracy 0.46596858638743455
correlation -0.04181174062032049
confusion
[[186 246]
 [264 259]]
remove all results
backtest on SOLUSDT
selected factors:
- weilun01              with number of config: 4
- weilun02              with number of c

factors:  71%|███████   | 51/72 [02:29<00:16,  1.28it/s]/Users/user/Desktop/stock/CryptoPredictor/pipeline.py:116: RankWarning: Polyfit may be poorly conditioned
  signal_output = pd.Series([fctr.Gen(row) for row in X], index=index, name=str(fctr))
/Users/user/Desktop/stock/CryptoPredictor/pipeline.py:116: RankWarning: Polyfit may be poorly conditioned
  signal_output = pd.Series([fctr.Gen(row) for row in X], index=index, name=str(fctr))
factors: 100%|██████████| 72/72 [03:17<00:00,  2.75s/it]


              train corr  test corr   weights
Slope_300      -0.127613   0.152927 -0.002049
weilun03_400   -0.098675  -0.001498 -0.066430
weilun04_400   -0.084938   0.133526 -0.052825
Slope_100      -0.042631   0.180088 -0.001352
weilun04_100   -0.038538   0.267417 -0.078209
...                  ...        ...       ...
weilun03_200    0.059871   0.129424  0.031684
Skewness_200    0.070651   0.063337 -0.010889
Skewness_20     0.076459   0.088757 -0.004289
Skewness_100    0.159222  -0.042206  0.034568
Skewness_50     0.167856  -0.166943  0.008687

[70 rows x 3 columns]

train result:
r2_score 0.14850420396682584
accuracy 0.6306282722513089
correlation 0.3863609036842063
confusion
[[1445  660]
 [ 751  964]]

test result:
r2_score -0.6763996757854553
accuracy 0.4408376963350785
correlation -0.16481718242318985
confusion
[[272 138]
 [396 149]]
remove all results
backtest on XRPUSDT
selected factors:
- weilun01              with number of config: 4
- weilun02              with number of con

factors:  71%|███████   | 51/72 [02:48<00:16,  1.29it/s]/Users/user/Desktop/stock/CryptoPredictor/pipeline.py:116: RankWarning: Polyfit may be poorly conditioned
  signal_output = pd.Series([fctr.Gen(row) for row in X], index=index, name=str(fctr))
factors: 100%|██████████| 72/72 [03:35<00:00,  3.00s/it]


              train corr  test corr   weights
Slope_300      -0.157600   0.066897 -0.000019
Skewness_200   -0.101088  -0.143228 -0.015038
Slope_200      -0.092717  -0.061738  0.000007
Slope_50       -0.062559   0.090530 -0.000021
weilun05_20    -0.059131   0.022757 -0.004085
...                  ...        ...       ...
Slope_3         0.038245   0.027294  0.000002
weilun03_100    0.047032   0.070070  0.000296
weilun03_300    0.058926  -0.166593  0.003670
weilun03_400    0.065761  -0.132625  0.029970
Slope_20        0.097313   0.133600  0.000016

[70 rows x 3 columns]

train result:
r2_score 0.13049832988793764
accuracy 0.6193717277486911
correlation 0.36235828388321706
confusion
[[1246  772]
 [ 682 1120]]

test result:
r2_score -1.0964890276391532
accuracy 0.5225130890052356
correlation 0.10280742651135842
confusion
[[317 156]
 [300 182]]
remove all results
backtest on DOGEUSDT
selected factors:
- weilun01              with number of config: 4
- weilun02              with number of co

factors: 100%|██████████| 72/72 [03:11<00:00,  2.65s/it]


                    train corr  test corr   weights
Slope_300            -0.110853  -0.070658 -0.000004
weilun05_200         -0.094602  -0.099493 -0.005179
Skewness_200         -0.086162  -0.170301 -0.000688
weilun03_400         -0.082225  -0.232820 -0.036576
weilun03_300         -0.078207  -0.332379 -0.024030
...                        ...        ...       ...
CrossRSI 10 30 10     0.078610  -0.039118  0.006652
CrossRSI 10 20 10     0.078751  -0.043606 -0.002684
CrossRSI 5 15 10      0.078958  -0.043529  0.002090
CrossRSI 3 8 10       0.080414  -0.039687  0.012735
CrossRSI 20 50 10     0.081175  -0.012668  0.010989

[70 rows x 3 columns]

train result:
r2_score 0.11787337855473312
accuracy 0.5992146596858638
correlation 0.3453406086397738
confusion
[[1287  802]
 [ 729 1002]]

test result:
r2_score -1.3407781116808333
accuracy 0.4617801047120419
correlation 0.15732851218560567
confusion
[[339 123]
 [391 102]]


In [6]:
import pandas as pd
targets = ["BTCUSDT", "ETHUSDT","BNBUSDT", "SOLUSDT", "XRPUSDT", "DOGEUSDT"]

results = []

for t in targets:
    df = pd.read_csv(f"/Users/user/Desktop/stock/CryptoPredictor/report/{t}_1000_72.csv")
    df.iloc[:,0] = df.iloc[:,0] + f"_{t}"
    results.append(df)

df = pd.concat(results, axis=0)
df = df.set_index("Unnamed: 0", drop=True)

In [ ]:
selected_index_long = df.index.str.contains("200|300|400")
selected_index_mid = df.index.str.contains("30|50|100")
selected = df[selected_index_long]

print("long term signals:", selected[["train corr","test corr"]].corr().iloc[0,1])
selected = df[selected_index_mid]
print("mid term signals:", selected[["train corr","test corr"]].corr().iloc[0,1])
selected = df[~selected_index_mid & ~selected_index_long]
print("other signals:", selected[["train corr","test corr"]].corr().iloc[0,1])

0.25979880001048805
-0.10437300254039543
-0.12413281555807311


In [30]:
selected

train corr  test corr   weights
Unnamed: 0                                                  
Skewness_10_BTCUSDT           -0.046608   0.108589 -0.003842
Cross_10_20_BTCUSDT           -0.044905   0.097156 -0.015620
Cross_5_15_BTCUSDT            -0.035306   0.116853 -0.014548
Skewness_5_BTCUSDT            -0.028203   0.073077 -0.000392
Skewness_3_BTCUSDT            -0.020712   0.018862 -0.000766
...                                 ...        ...       ...
CrossRSI 10 20 10 _DOGEUSDT    0.078751  -0.043606 -0.002684
CrossRSI 5 15 10 _DOGEUSDT     0.078958  -0.043529  0.002090
CrossRSI 3 8 10 _DOGEUSDT      0.080414  -0.039687  0.012735
weilun02_0.4_3_DOGEUSDT             NaN        NaN  0.000000
weilun02_0.2_3_DOGEUSDT             NaN        NaN  0.000000

[252 rows x 3 columns]

In [116]:
df_eth = pd.read_csv(f"/Users/user/Desktop/stock/CryptoPredictor/report/{t}_1000_72.csv")

In [19]:
# df_eth = df_eth.set_index("Unnamed: 0")
consider = df_eth[df_eth.index.str.contains("200|300|400")]
consider[["train corr", "test corr"]].corr()

NameError: name 'df_eth' is not defined

## Backtest with only 400 timeframe signals.


remove all results
backtest on BTCUSDT
selected factors:
- weilun01              with number of config: 4
- weilun02              with number of config: 12
- weilun03              with number of config: 9
- weilun04              with number of config: 10
- weilun05              with number of config: 10
total number of factors: 45
factors: 100%|██████████| 45/45 [02:42<00:00,  3.60s/it]
              train corr  test corr   weights
weilun04_400   -0.037646  -0.054631  0.006058
weilun01_400   -0.016535   0.230540 -0.003276
weilun05_400    0.024863  -0.010956 -0.000094
weilun03_400    0.107799   0.125773  0.027973

train result:
- r2_score 0.012930839909615521
- accuracy 0.5586387434554974
- correlation 0.11392766769588172
- confusion
[[1297  718]
 [ 968  837]]

test result:
- r2_score -0.08294181360991826
- accuracy 0.45968586387434557
- correlation 0.007378769215386666
- confusion
[[206 194]
 [322 233]]
remove all results
backtest on ETHUSDT
selected factors:
- weilun01              with number of config: 4
- weilun02              with number of config: 12
- weilun03              with number of config: 9
- weilun04              with number of config: 10
- weilun05              with number of config: 10
total number of factors: 45
factors: 100%|██████████| 45/45 [02:51<00:00,  3.82s/it]
              train corr  test corr   weights
weilun03_400   -0.084763  -0.161290 -0.028503
weilun04_400   -0.031046   0.062462 -0.014404
weilun01_400   -0.002185   0.136790  0.004942
weilun05_400   -0.001679  -0.121260  0.001190

train result:
- r2_score 0.009817225064164492
- accuracy 0.4905759162303665
- correlation 0.09929532773109621
- confusion
[[1194  711]
 [1235  680]]

test result:
- r2_score 0.020011506470013818
- accuracy 0.6041884816753926
- correlation 0.20350051398239516
- confusion
[[312 146]
 [232 265]]
remove all results
backtest on BNBUSDT
selected factors:
- weilun01              with number of config: 4
- weilun02              with number of config: 12
- weilun03              with number of config: 9
- weilun04              with number of config: 10
- weilun05              with number of config: 10
total number of factors: 45
factors: 100%|██████████| 45/45 [03:03<00:00,  4.07s/it]
              train corr  test corr   weights
weilun03_400   -0.087097  -0.135193 -0.033242
weilun04_400   -0.005180   0.264911 -0.020121
weilun05_400    0.004274  -0.169359  0.000319
weilun01_400    0.010556   0.178717  0.003763

train result:
- r2_score 0.01365588275496521
- accuracy 0.5246073298429319
- correlation 0.11744485538630751
- confusion
[[1333  657]
 [1159  671]]

test result:
- r2_score -0.02341827427305687
- accuracy 0.5403141361256545
- correlation 0.033498623384041776
- confusion
[[169 263]
 [176 347]]
remove all results
backtest on SOLUSDT
selected factors:
- weilun01              with number of config: 4
- weilun02              with number of config: 12
- weilun03              with number of config: 9
- weilun04              with number of config: 10
- weilun05              with number of config: 10
total number of factors: 45
factors: 100%|██████████| 45/45 [02:51<00:00,  3.82s/it]
              train corr  test corr   weights
weilun03_400   -0.098675  -0.001498 -0.038478
weilun04_400   -0.084938   0.133526 -0.030540
weilun05_400   -0.013357  -0.166742  0.001592
weilun01_400   -0.003660   0.223426  0.017616

train result:
- r2_score 0.01905839635525375
- accuracy 0.5379581151832461
- correlation 0.13814928722965733
- confusion
[[1336  769]
 [ 996  719]]

test result:
- r2_score -0.25884895663765595
- accuracy 0.4178010471204188
- correlation 0.12236759549196546
- confusion
[[397  13]
 [543   2]]
remove all results
backtest on XRPUSDT
selected factors:
- weilun01              with number of config: 4
- weilun02              with number of config: 12
- weilun03              with number of config: 9
- weilun04              with number of config: 10
- weilun05              with number of config: 10
total number of factors: 45
factors: 100%|██████████| 45/45 [02:42<00:00,  3.61s/it]
              train corr  test corr   weights
weilun05_400   -0.045720  -0.018406 -0.003198
weilun04_400   -0.034918  -0.050632  0.008006
weilun01_400   -0.022897  -0.037520 -0.008091
weilun03_400    0.065761  -0.132625  0.023706

train result:
- r2_score 0.008455101484022287
- accuracy 0.5293193717277487
- correlation 0.09201690602597762
- confusion
[[ 937 1081]
 [ 717 1085]]

test result:
r2_score -0.07967927147738485
accuracy 0.4324607329842932
correlation -0.10392607524359972
confusion
[[241 232]
 [310 172]]
remove all results
backtest on DOGEUSDT
selected factors:
- weilun01              with number of config: 4
- weilun02              with number of config: 12
- weilun03              with number of config: 9
- weilun04              with number of config: 10
- weilun05              with number of config: 10
total number of factors: 45
factors: 100%|██████████| 45/45 [02:55<00:00,  3.89s/it]
              train corr  test corr   weights
weilun03_400   -0.082225  -0.232820 -0.048693
weilun04_400   -0.056034   0.022431 -0.043784
weilun01_400   -0.039267   0.051981  0.004139
weilun05_400    0.015061  -0.072862 -0.001492

train result:
- r2_score 0.02197005234478988
- accuracy 0.5845549738219895
- correlation 0.14862325119248737
- confusion
[[1376  713]
 [ 874  857]]

test result:
- r2_score -0.059643919688049385
- accuracy 0.5026178010471204
- correlation 0.28100881723958426
- confusion
[[427  35]
 [440  53]]